### PR TITLE
Complete Primitive Data Structures

### LINKED ISSUE #1: [bounty 7] Make Transaction class to represent transaction
- incapsulate Inputs and Outputs
- make it signable
- get rid of `Signer`

### PR DESCRIPTION
**Pull request by troggy**
- [x] make transaction a proper object holding Inputs and Outputs
- [x] serialize on request with `.toRaw()`
- [x] got rid of `Signer` object

Resolves:
- https://github.com/parsec-labs/PIPs/wiki/Bounty_007
- #1 
- #2

### CODE REVIEW

File: lib/block.js [hunk 1]
```diff
@@ -9,122 +9,122 @@ export default class Block {
...
+    const metadataAndTxLength = 9 + txData.length;
+    // align to 32 bytes, so that e.g. 38 bytes will be stored in 64 bytes buffer
+    const doubleWordAlignedLength = metadataAndTxLength + (32 - (metadataAndTxLength % 32));
+    const txPayload = Buffer.alloc(doubleWordAlignedLength);
     // tx length
-    payload.writeUInt32BE(txData.length, 0);
-    // offset
-    let offset = ((txData.length % 32) <= 19) ? 32 - (txData.length % 32) : (32 - (txData.length % 32)) + 32;
-    offset += 96;
-    payload.writeUInt8(offset, 0);
```
Comment (troggy) [inline lib/block.js]: @johannbarbie this writes to offset 0 and txData length also written to offset 0. Was it an error?

Comment (johannbarbie) [reply lib/block.js]: we use only 2 bytes of the length, so we can overwrite the first 2 bytes.

Comment (troggy) [reply lib/block.js]: stupid question.. the answer is in the code

File: lib/block.js [hunk 2]
```diff
@@ -91,8 +97,8 @@ export default class Block {
     slices.push(bufferToHex(this.s));
 
     // Slices with tx metadata, v-sig and tx data
-    // metadata size: 4 bytes tx length + 4 bytes tx index + 1 byte v-sig = 9 bytes
-    const metadataAndTxLength = 9 + txData.length;
+    // metadata size is 13 bytes (see proof structure above)
```
Comment (johannbarbie) [inline lib/block.js]: interesting that all the test passed, and all the proofs in the ParsecBridge didn't fail

Comment (troggy) [reply lib/block.js]: yeah.. I writing some unit tests to check this boundary conditions (e.g. tx size of 32 + 18)
